flow/project/cli/config/composer.go

```diff
@@ -0,0 +1,220 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+// ErrDoesNotExist is error to be returned when config file does not exists
+var ErrDoesNotExist = errors.New("project config file does not exist")
+
+// Exists checks if file exists on the specified path
+func Exists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// Parser is interface for any configuration format parser to implement
+type Parser interface {
+	Serialize(*Config) ([]byte, error)
+	Deserialize([]byte) (*Config, error)
+	SupportsFormat(string) bool
+}
+
+// ConfigParsers lsit of all configuration parsers
+type ConfigParsers []Parser
+
+// FindForFormat finds a parser that can parse a specific format based on extension
+func (c *ConfigParsers) FindForFormat(extension string) Parser {
+	for _, parser := range *c {
+		if parser.SupportsFormat(extension) {
+			return parser
+		}
+	}
+
+	return nil
+}
+
+// Loader contains actions for composing and modification of configuration
+type Loader struct {
+	af               *afero.Afero
+	configParsers    ConfigParsers
+	composedMultiple bool
+	composedFromFile map[string]string
+}
+
+// NewLoader creates a new loader
+func NewLoader(filesystem afero.Fs) *Loader {
+	af := &afero.Afero{Fs: filesystem}
+	return &Loader{
+		af:               af,
+		composedFromFile: map[string]string{},
+	}
+}
+
+// AddConfigParser adds new parssers for configuration
+func (c *Loader) AddConfigParser(format Parser) {
```

Comment:
```suggestion
func (l *Loader) AddConfigParser(format Parser) {
```

Same for the rest of the `Loader` methods